spec/inertia/conditional_sharing_spec.rb

```diff
@@ -1,28 +1,27 @@
+# Specs as documentation. Per-action shared data isn't explicity supported,
+# but it can be done by referencing the action name in an inertia_share block.
 RSpec.describe "conditionally shared data in a controller", type: :request do
-  context "when there is conditional data inside inertia_share" do
-    it "does not leak data between requests" do
-      get conditional_share_index_path, headers: {'X-Inertia' => true}
-      expect(JSON.parse(response.body)['props'].deep_symbolize_keys).to eq({
-                                                                             index_only_prop: 1,
-                                                                             normal_shared_prop: 1,
-                                                                           })
-
-      # NOTE: we actually have to run the show action twice since the new implementation
-      # sets up a before_action within a before_action to share the data.
-      # In effect, that means that the shared data isn't rendered until the second time the action is run.
-      get conditional_share_show_path, headers: {'X-Inertia' => true}
+  context "when there is data inside inertia_share only applicable to a single action" do
+    it "does not leak the data between requests" do
       get conditional_share_show_path, headers: {'X-Inertia' => true}
       expect(JSON.parse(response.body)['props'].deep_symbolize_keys).to eq({
-                                                                             normal_shared_prop: 1,
-                                                                             show_only_prop: 1,
-                                                                             conditionally_shared_show_prop: 1,
-                                                                           })
+        normal_shared_prop: 1,
+        show_only_prop: 1,
+        conditionally_shared_show_prop: 1,
+      })
 
       get conditional_share_index_path, headers: {'X-Inertia' => true}
-      expect(JSON.parse(response.body)['props'].deep_symbolize_keys).to eq({
-                                                                             index_only_prop: 1,
-                                                                             normal_shared_prop: 1,
-                                                                           })
+      expect(JSON.parse(response.body)['props'].deep_symbolize_keys).not_to include({
+        conditionally_shared_show_prop: 1,
+      })
+    end
+  end
+
+  context "when there is conditional data shared via before_action" do
```

Comment:
Freezing the shared data prevents the `before_action` conditional sharing hack I cooked up! It's a really nice guard rail.